### DISCUSSION TITLE
Question: recommended project structure

### DISCUSSION THREAD

**Discussion by emilecaron** [riverqueue/river#671] [Q&A]
Hello,

I wonder if there is a recommended project structure when you have more than a single package.
Client setup needs to import all the workers in various packages.
But these jobs may require the client and the other jobs to be able to insert them.

Is there a recommended way to structure the project in order to avoid cyclical dependencies ? 

Thanks

**Comment by assembly-winston** [comment]
I usually use a `riverschema` package to specify args, a `workers` folder with subfolders/packages for each worker. I find this generally avoids most cyclical dependencies.
